### PR TITLE
Initial dev database setup

### PR DESCRIPTION
**Pull request by jniemii**
Add a docker-compose config for a development PostgreSQL database and connect to it with a pg Pool at startup.

For development, the database configuration is shared by both backend and docker through the .env file. People with existing .env need to update it according to .env-sample

Does not currently retain any state after shutdown.

### CODE REVIEW

**Review by samikuikka — APPROVED** [review]
I wish it works on Windows too

**Comment by jniemii** [comment]
> I wish it works on Windows too

Should now work on Windows

File: backend/package.json [hunk 1]
```diff
@@ -4,12 +4,16 @@
...
         "tsc": "tsc",
-        "dev": "ts-node-dev src/index.ts",
+        "dev": "run-p start-dev-database start-dev",
         "lint": "eslint --ext .ts .",
         "build": "tsc",
-        "start": "node build/index.js"
+        "start": "node build/index.js",
+        "start-dev": "ts-node-dev src/index.ts",
+        "start-dev-database": "(docker-compose down || echo skipped docker-compose down) && docker-compose up",
+        "copy-sample-env": "cpy .env-sample . --no-overwrite --rename=.env || echo skipped copy-sampe-env"
```
Comment (KorpelainenTommi) [inline backend/package.json]: Typo in line 16 "echo skipped copy-**sampe**-env"

**Review by KorpelainenTommi — COMMENTED** [review]
Tested and this should work on Windows as long as docker is available (eg. found in PATH, docker desktop running etc.)

One minor typo. Some _Javascriptierds_ also dislike the use of `.then` and `.catch`

File: backend/src/db/pool.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Pool } from 'pg';
+
+const pool = new Pool({
+    user: process.env.POSTGRES_USER,
+    host: 'localhost',
```
Comment (hmalmsten) [inline backend/src/db/pool.ts:5]: POSTGRES_URL is defined in .env so use it here or delete it from .env

File: backend/.env-sample [hunk 3]
```diff
@@ -0,0 +1,7 @@
+DUMMYURL=http://localhost:8080
+POSTGRES_USER=aalto_user
+# change and uncomment or use copy-sample-env script
+#POSTGRES_PASSWORD=CHANGEME
```
Comment (jaakkoha-bytecraft) [inline backend/.env-sample]: I'd personally use the same password for everyone for dev setup -- it makes things simpler, and having a different password for local envs doesn't bring any security benefits.

**Pull request merged**